### PR TITLE
Organize subject components

### PR DESCRIPTION
**Pull request by parrish**
The subject index page was getting pretty huge, so before I added anything else to it, I wanted to try to start breaking out some components.

I also tried to clean up and refactor a bit too.

### CODE REVIEW

File: app/subjects/comment-form.cjsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    <Link to="project-classify" params={{owner, name}}>{text}</Link>
+
+  popup: ->
+    alert (resolve) -> <SignInPrompt onChoose={resolve} />
+
+  loginPrompt: ->
+    <p>
+      Please{' '}
+      <button className="link-style" type="button" onClick={@popup}>
+        sign in
```
Comment (aweiksnar) [inline app/subjects/comment-form.cjsx]: Can you add in a space here?

**Comment by aweiksnar** [comment]
Super diff! Thanks! 

This page was in need of some attention. I had written it in a bit of a rush on that late-night launch...

**Comment by parrish** [comment]
Updated, thanks!

**Comment by aweiksnar** [comment]
Looks great :+1:

**Pull request merged**